libkatana_python_native/include/katana/python/CythonIntegration.h

```diff
@@ -8,12 +8,27 @@
 ///
 /// The interface provided by both pybind11 and Cython objects entirely call
 /// via Python to avoid the need to have an ABI between pybind11 and Cython
-/// code. The interface is:
+/// code. The interface is (not every class will implement every method):
 ///
-/// * an instance property __katana_address__ which returns the address of the
+/// - an instance property __katana_address__ which returns the address of the
 ///   underlying C++ object as an int.
-/// * a static method _make_from_address which takes the C++ object pointer as
-///   an int, and returns a new Python object wrapping it.
+/// - a static method _make_from_address which takes T* as
+///   an int and an owner object, and returns a new Python object wrapping it.
+///   Does not take ownership, but will attempt to keep the owner alive as long
+///   as the wrapper is alive.
+/// - a static method _make_from_address_own which takes T*
+///   as an int, and returns a new Python object wrapping it. Takes
+///   ownership.
+/// - a static method _make_from_address_shared which takes shared_ptr<T>*
+///   as an int, and returns a new Python object wrapping it. Creates a new
+///   shared_ptr internally.
+/// - a static method _make_from_address_unique which takes unique_ptr<T>*
+///   as an int, and returns a new Python object wrapping it. Moves out of
+///   the unique_ptr.
```

Comment:
Wdym by "moves out of the unique_ptr"? Does this mean we don't take ownership, rather make a copy?

Reply:
I mean it takes ownership of the referenced object via `std::move`. Bad wording sorry.